.devcontainer/configure.py

```diff
@@ -0,0 +1,158 @@
+
+CURRENT_DIR = os.path.dirname(
+    os.path.abspath(inspect.getfile(inspect.currentframe())))
+
+DOCKER_IMAGE_SHORTNAME_DICT = {
+    "base": {
+        "latest": "base-bleeding-edge",
+        "stable": "base"
+    },
+    "nvidia": {
+        "latest": "nvidia-bleeding-edge",
+        "stable": "nvidia"
+    }
+}
+
+
+def run_shell(cmd, allow_non_zero=False, stderr=None):
+  cmd = shlex.split(cmd)
+  if stderr is None:
+    stderr = sys.stdout
+  if allow_non_zero:
+    try:
+      output = subprocess.check_output(cmd, stderr=stderr)
+    except subprocess.CalledProcessError as e:
+      output = e.output
+  else:
+    output = subprocess.check_output(cmd, stderr=stderr)
+  return output.decode('UTF-8').strip()
+
+
+def is_nvidia_gpu_available():
+  """This function verifies NVIDIA Docker runtime is installed and 
+  available. It also checks NVIDIA GPU are available and the driver
+  is installed."""
+
+  docker_executable = which('docker')
+  if docker_executable is None:
+    raise RuntimeError("Docker is not installed or in the user path.")
+
+  command_str = f"{docker_executable} info --format '{{{{json .}}}}'"
+  data = json.loads(run_shell(command_str))
+
+  if "nvidia" not in data["Runtimes"].keys():
+    return False, "NVIDIA Docker Runtime is not available"
+
+  nvidia_smi_executable = which('nvidia-smi')
+  if nvidia_smi_executable is None:
+    return False, "NVIDIA Driver is not installed or not in the user path."
+
+  command_str = f"{nvidia_smi_executable} --query-gpu=gpu_name --format=csv"
+
+  if run_shell(command_str).splitlines()[1:]:  # Skip header
+    return True, None
+  else:
+    return False, "No NVIDIA GPU was found on this system."
+
+
+def get_input(question, default_answer='', accepted_answers=None):
+  accepted_answers = [x.strip().lower() for x in accepted_answers]
+
+  while True:
+
+    try:
+      answer = input(f"{question} ")  # pylint: disable=bad-builtin
+    except EOFError:
+      answer = default_answer
+    answer = (answer or default_answer).strip().lower()
+
+    if answer not in accepted_answers:
+      print(
+          f"Unsupported answer received: {answer}. Expected: {accepted_answers}"
+      )
+      continue
+
+    break
+
+  return answer.strip().lower()
+
+
+if __name__ == "__main__":
+
+  if os.path.isfile(os.path.join(CURRENT_DIR, "docker-compose.yml")):
+    if get_input(
+        "A `docker-compose.yml` already exists. Are you certain you want to overwrite it [y/N]?",
+        default_answer="n",
+        accepted_answers=["y", "n"]) == "n":
+      sys.exit(1)
+
+  with open(os.path.join(CURRENT_DIR, "docker-compose.base.yml")) as f:
+    docker_config = json.load(f)
+
+  use_official_img = get_input(
+      "Do you wish to use the official prebuild development containers [Y/n]?",
+      default_answer="y",
+      accepted_answers=["y", "n"]) == "y"
+
+  use_bleeding = get_input(
+      "Do you wish to use the bleeding-edge container (might be unstable) [Y/n]?",
```

Comment:
These actually aren't really any more unstable. They're just the latest version of things we support (bleeding edge is a bit hyperbolic, but naming is hard) as opposed to the *oldest* version of what we support. I think users are likely to have a nicer time on the more modern images, so I think we shouldn't push them away from those. In fact, I think I would just always use the bleeding-edge images and not even ask about the other ones

Reply:
Let's offer the possibility given how simple it is, does "legacy" container for the "non bleeding-edge" sound fair to you ?

Legacy vs Latest ?

Reply:
I applied your comment and fixed it ;)